front/tests/pages/op-simulation-results-page-model.ts

```diff
@@ -0,0 +1,52 @@
+import { type Locator, type Page } from '@playwright/test';
+
+class OpSimulationResultPage {
+  readonly page: Page;
+
+  private readonly speedSpaceChartSettingsButton: Locator;
+
+  private readonly speedSpaceChartSettingsCheckboxes: Locator;
+
+  private readonly speedSpaceChartCloseSettingsButton: Locator;
+
+  private readonly speedSpaceChartCheckboxItems: Locator;
+
+  readonly speedSpaceChartTabindexElement: Locator;
+
+  constructor(page: Page) {
+    this.page = page;
+    this.speedSpaceChartSettingsButton = page.locator('.interaction-button.elipsis-button');
+    this.speedSpaceChartSettingsCheckboxes = page.locator(
+      '#settings-panel .selection input[type="checkbox"]'
+    );
+    this.speedSpaceChartCloseSettingsButton = page.locator('#close-settings-panel');
+    this.speedSpaceChartCheckboxItems = page.locator('#settings-panel .selection .checkmark');
+    this.speedSpaceChartTabindexElement = page.locator(
+      '#container-SpeedSpaceChart > div[tabindex="0"]'
+    );
+  }
+
+  private async openSettingsPanel(): Promise<void> {
+    await this.speedSpaceChartSettingsButton.click();
+  }
+
+  private async closeSettingsPanel(): Promise<void> {
+    await this.speedSpaceChartCloseSettingsButton.click();
+  }
+
+  // Ensures all checkboxes in the settings panel are checked.
+  async selectAllSpeedSpaceChartCheckboxes(): Promise<void> {
+    await this.openSettingsPanel();
+
+    const checkboxCount = await this.speedSpaceChartSettingsCheckboxes.count();
+
+    for (let checkboxIndex = 0; checkboxIndex < checkboxCount; checkboxIndex += 1) {
+      await this.speedSpaceChartCheckboxItems.nth(checkboxIndex).setChecked(true);
+    }
```

Comment:
Continuing discussion from https://github.com/OpenRailAssociation/osrd/pull/9829#discussion_r1868067168

So, can this be replaced with `this.speedSpaceChartSettingsCheckboxes.setChecked(true)`? Or does this not work somehow?

Reply:
`setChecked()` method checks or unchecks an element can't be done for all the checkoboxes

Reply:
Done